openff/evaluator/backends/dask_kubernetes.py

```diff
@@ -0,0 +1,484 @@
+                "resources": {"requests": {"storage": f"{storage_tb}Ti"}},
+            },
+        }
+        return spec
+
+    def _to_volume_spec(self):
+        return {
+            "name": self.name,
+            "persistentVolumeClaim": {"claimName": self.name},
+        }
+
+    @requires_package("kubernetes")
+    def _to_volume_k8s(self):
+        from kubernetes import client
+
+        return client.V1Volume(
+            name=self.name,
+            persistent_volume_claim=client.V1PersistentVolumeClaimVolumeSource(
+                claim_name=self.name
+            ),
+        )
+
+
+class KubernetesEmptyDirVolume(BaseKubernetesVolume):
+    """A helper class for specifying Kubernetes emptyDir volumes."""
+
+    read_only: bool = Field(
+        False, description="Whether the volume should be read-only."
+    )
+
+    def _to_volume_spec(self):
+        return {
+            "name": self.name,
+            "emptyDir": {},
+        }
+
+    @requires_package("kubernetes")
+    def _to_volume_k8s(self):
+        from kubernetes import client
+
+        return client.V1Volume(
+            name=self.name,
+            empty_dir=client.V1EmptyDirVolumeSource(),
+        )
+
+
+class BaseDaskKubernetesBackend(BaseDaskBackend):
+
+    def __init__(
+        self,
+        gpu_resources_per_worker=None,
+        cpu_resources_per_worker=None,
+        cluster_name="openff-evaluator",
+        cluster_port=8786,
+        disable_nanny_process=False,
+        image: str = "ghcr.io/lilyminium/openff-images:evaluator-0.4.10-kubernetes-dask-v0",
+        namespace: str = "openforcefield",
+        env: dict = None,
+        secrets: list[KubernetesSecret] = None,
+        volumes: list[KubernetesPersistentVolumeClaim] = None,
+        cluster_kwargs: dict = None,
+        annotate_resources: bool = False,
+        include_jupyter: bool = False,
+    ):
+        default_resources = None
+        other_resources = {}
+        if gpu_resources_per_worker is not None:
+            # preference gpu resources over cpu
+            assert isinstance(gpu_resources_per_worker, PodResources)
+            default_resources = gpu_resources_per_worker
+            other_resources["cpu"] = cpu_resources_per_worker
+        elif cpu_resources_per_worker is not None:
+            assert isinstance(cpu_resources_per_worker, PodResources)
+            default_resources = cpu_resources_per_worker
+        else:
+            raise ValueError(
+                "Either gpu_resources_per_worker or cpu_resources_per_worker must be specified."
+            )
+
+        super().__init__(
+            default_resources._minimum_number_of_workers,
+            default_resources,
+        )
+
+        self._cluster_name = cluster_name
+        self._cluster_port = cluster_port
+        self._namespace = namespace
+        self._annotate_resources = annotate_resources
+        self._image = image
+        self._other_resources = other_resources
```

Comment:
Since this is currently only GPU/CPU we could also just skip this and set `self._cpu_resources_per_worker=...` etc.